### DISCUSSION TITLE
Manual Leveling Bed w/PROBE to fine tune (Chasing Zeros)

### DISCUSSION THREAD

**Discussion by K2Kevin** [Jyers/Marlin#1882] [General Discussion]
Using the knobs to complete your initial leveling and fine tuning your bed level after viewing your Mesh results.  Im always chasing zeros in my bed leveling efforts.  I generally settle on .01 - .05.  I started using the option to use PROBE in the manual leveling menu and was able to achieve almost "perfection" where most of the beds printable area is all ZEROS or .001.  Using the probe to fine tune your bed level is a great feature for those chasing Zeros. Had to change the default scale (Visualizer looked flat) of the Bed visualizer to even see the bed differences.

![Screen Shot 2022-03-02 at 2 04 51 PM](https://user-images.githubusercontent.com/27362912/156441842-b99bdce8-9306-40ba-96f3-0cf30025eb9a.jpg)

**Comment by GeoStork** [comment]
Is there a Wiki regarding how to us or how this manual leveling is supposed to work?  I just compiled the "i" version and explored the manual level options and clicked on "with probe".  the machine cycled through each corner, probing, then stoped and did nothing further.  Certainly user error on my part as I had no idea what I was doing.   I proceeded normally to set my z-offset with a piece of paper and then initiated an ABL, saved the mesh and proceeded to print.  Didn't have time to go back and play with the corner adjustments, etc.

**Reply by BlueWings172** [reply]
@GeoStork 
I'm not sure if there is a guide for the manual leveling by probe and I also came across this function by chance and I love it.

But we can make our own guide and help others. It is quite simple and short:

So the purpose of manual leveling with probe is to level the bed arcuately without using the paper and by relying on the readings of the BLtouch or CRTouch.

1- Go to` Prepare > Manual Leveling` (printer might home itself if it is not homed already)
2- Activate `With Probe` checkbox
3- Printer will probe 4 corner. 
4- Printer display shows the list of all corners and center. Select `Bottom Left`. the printer will move to that corner but won't probe it yet.
5- Printer will ask `Confirm` or `Cancel`. `Confirm` will probe the corner and `Cancel` will return to the list of corners. Select `Confirm`.
6- Printer will probe the corner twice and display how high or low this corner is. I think this is the part where people might miss seeing the reading because it is so small. I hope it can be in red font or something. Please see the photo below for where the reading will show on the screen.
7- Adjust the knob of that corner accordingly. If the printer says `High`, turn knob right (counterclockwise), if it says `Low`, turn the knob left (clockwise). Make small adjustments and after the first couple of times, you would understand how much you turn will impact the reading. 
8- After you turned the knob, Select `Confirm` from the display again to make another reading and then adjust the know accordingly. It may take 3 or 4 times until you get that corner leveled. Adjusting one corner affects the others corners so multiple cycles are needed. What I look for is a reading less than 0.01. So **a reading of 0.005 or 0.008 is perfect**. I prefer to have my corners on the High side because the center of my bed is lower than the corners. So anything between 0.00 to 0.009 High is perfect for me.
9- Once satisfied with the adjustment of Bottom Left, select `Top Left `and then `Confirm`, wait for the reading, adjust the knob, take another reading and move on to next corner once you are happy with it.
10- I usually keep moving from corner to the next until I reach the last corner I adjusted which means that I completed a cycle of 4 corners without needing to adjust anything. At that point I am done. Usually this takes me about 20 minutes and results are worth it.
11- Needless to say that there is nothing to save here because this is a mechanical tuning.

**Tips:**
1- I found that after the first cycle (4 corners), it is better to probe and adjust opposing corners rather than corners in the same sequence as the screen.
2- While it seems that heating the bed is a good idea, I read somewhere that Bltouch accuracy can be affected by heated bed. I tried heated bed at 65c, 40c and cold and did not notice a difference. 
3- If you notice that the probe readings are not consistent, It is a good Idea to run M48 V4 command in Octoprint Terminal or Pronterface. This command will run a probe repeatability test and give you a standard deviation result. It is recommended to run it more than once (each test will probe 10 times I think). Anything around or below 0.005 means that your probe is working well.

I am no expert and others are welcome to share their own tips and tricks.


![20220317_132214](https://user-images.githubusercontent.com/65966061/158801270-65cd92f1-4460-4ed2-b6d1-f6ee8bfdb585.jpg)

**Comment by PapsDizzy** [comment]
@K2Kevin wich change you have made in bed visualizer?

**Reply by K2Kevin** [reply]
I changed the visualization values for z Limits and colorscale.  The bed is so level, the default was just all green and flat.

Z Limits: auto, auto
Colorscale: RdBu

**Comment by wfdTamar** [comment]
I've just worked this out before seeing this on my new Ender 3 V2 (my first printer). 

One more point is the rear left (why do they call it top or bottom?) screw holds the hot beds electrical cable, so I start with that done up fairly tight, manually set the height of the nozzle paper thickness above that, then manually set it the same height above the other corners.

Work out the Z height (difference between probe tip and nozzle tip heights) and enter it in the settings. It'll be around 3mm with the BL Touch factory mount (but measure it to be sure).

Then work out the Probe Offset measurements. These will be about -45mm & -5mm (but again measure your machine). Enter those in the settings (not east to find as they're not called Probe Offset!).

Then do the manual adjustment to the bed as described by the OP.

If you have loaded a firmware with mesh setup do an automatic mesh scan and save it.

P.S. I'm a bit of a rookie, so this may not be right  :-)
